core/services/assets/AssetManifest.php

```diff
@@ -0,0 +1,329 @@
+ * @author  Brent Christensen
+ * @since   $VID:$
+ */
+class AssetManifest
+{
+
+    const ASSET_EXT_CSS      = '.css';
+
+    const ASSET_EXT_JS      = '.js';
+
+    const ASSET_EXT_PHP     = '.php';
+
+    const FILE_NAME         = 'asset-manifest.json';
+
+    const KEY_DEPENDENCIES  = 'dependencies';
+
+    const KEY_ENTRY_POINTS  = 'entrypoints';
+
+    const KEY_FILES         = 'files';
+
+    const KEY_VERSION       = 'version';
+
+
+    /**
+     * @var array
+     */
+    private $asset_files;
+
+    /**
+     * @var string
+     */
+    private $assets_namespace;
+
+    /**
+     * @var string
+     */
+    private $assets_path;
+
+    /**
+     * @var DomainInterface
+     */
+    protected $domain;
+
+    /**
+     * @var array
+     */
+    private $entry_points;
+
+    /**
+     * @var array
+     */
+    private $manifest;
+
+    /**
+     * @var string
+     */
+    private $manifest_path;
+
+    /**
+     * This is a list of known handles that are used for css.
+     * @var array
+     */
+    private $wp_css_handle_dependencies = [
+        'wp-components',
+        'wp-block-editor',
+        'wp-block-library',
+        'wp-edit-post',
+        'wp-edit-widgets',
+        'wp-editor',
+        'wp-format-library',
+        'wp-list-reusable-blocks',
+        'wp-nux',
+    ];
+
+
+    /**
+     * AssetManifest constructor.
+     *
+     * @param DomainInterface $domain
+     */
+    public function __construct(DomainInterface $domain)
+    {
+        $this->domain = $domain;
+    }
+
+
+    /**
+     * @return void
+     */
+    public function initialize()
+    {
+        if (! $this->manifest) {
+            $this->assets_namespace = $this->domain->assetNamespace();
+            $assets_path = $this->domain->distributionAssetsPath();
+            $this->assets_path = trailingslashit($assets_path);
+            $this->setManifestFilepath();
+            $this->loadManifest();
+            $this->getAssetFiles();
+            $this->getEntryPoints();
+        }
+    }
+
+
+    /**
+     * @param string $manifest_path
+     */
+    public function setManifestFilepath($manifest_path = '')
+    {
+        $manifest_path = $manifest_path ? $manifest_path : $this->assets_path . AssetManifest::FILE_NAME;
+        if (! is_readable($manifest_path)) {
+            throw new AssetManifestException(
+                sprintf(
+                    esc_html__(
+                        'Manifest file was not found or is not readable. Please try running "yarn dev" in the console and/or verify that "%1$s" exists and has valid permissions.',
+                        'event_espresso'
+                    ),
+                    $manifest_path
+                )
+            );
+        }
+        $this->manifest_path = $manifest_path;
+    }
+
+
+    /**
+     * @return void
+     */
+    private function loadManifest()
+    {
+        if (! $this->manifest) {
+            // TODO May be use WP File system? ¯\_(ツ)_/¯
```

Comment:
if that is battle-tested and is keeping up with latest PHP innovations, then why not

Reply:
this comment and line of code it refers to were written by Manzoor but I should likely take this advice

Reply:
I recently realized that WP File system is a problem if the server permissions are not properly set up. You may end up not being able to read the manifest file. It's fine to use direct method via PHP.